hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -92,7 +94,19 @@ public void mint(TokenRelationship treasuryRel, long amount) {
 		changeSupply(treasuryRel, +amount, INVALID_TOKEN_MINT_AMOUNT);
 	}
 
-	public OwnershipTracker mint(OwnershipTracker ownershipTracker, TokenRelationship treasuryRel, List<ByteString> metadata, final RichInstant creationTime) {
+	public void burn(final OwnershipTracker ownershipTracker, final TokenRelationship treasuryRelationship, final List<Long> serialNumbers){
+		validateTrue( type == TokenType.NON_FUNGIBLE_UNIQUE, INVALID_TRANSACTION_BODY, () ->
+				"Non fungible burn can be invoked only on Non fungible tokens!");
+		changeSupply(treasuryRelationship, serialNumbers.size() *-1, FAIL_INVALID);
+
+		for (final long serialNum : serialNumbers) {
+			ownershipTracker.add(id, OwnershipTracker.fromBurning(id, serialNum));
+			burnedUniqueTokens.add(new UniqueToken(id, serialNum));
+		}
+		treasury.setOwnedNfts(treasury.getOwnedNfts() - serialNumbers.size());
```

Comment:
nit: On the mint, we are incrementing for each metadata. I would say that there is not a big difference in the 2 approaches. I would like to say just to keep it consistent